### DISCUSSION TITLE
Help Request: [Unable to start back end dev server]

### DISCUSSION THREAD

**Discussion by Jellybeano** [TEAMMATES/teammates#11217] [Help requests] [answered]
- **Commit SHA:** 744b69a6339bd1dc568bdf2f1ebe99f13966d17e
- **Operating System:** Windows 10
- **JDK Version:** java version "1.8.0_51" 
- **Node.js Version:** v14.17.0   
- **Google Cloud SDK Version:** NIL
- **Firefox / Chrome Version (if applicable):** NIL

**What I was trying to do / What I need help with**

Was following the guide to manage the dev server when I ran into a problem when starting the backend server: 
![image](https://user-images.githubusercontent.com/65604398/122845424-556f8a00-d336-11eb-9fed-7c152a71831e.png)

Error 404 Not Found is seen when going to http://localhost:8080/ 

The developer environment has been set up following the guide, and there is a running database instance and a full-text search service instance.

**Attempts to resolve the problem**

After trying multiple things such as deleting all files and redoing the process of setting up, the problem remains. (and now the runDatastoreEmulator command does not even result in an OK at http://localhost:8484/ after a successful build)

Seeking some help on the issue, would be very much appreciated!

**Comment by damithc** [comment]
@TEAMMATES/active-mentors any thoughts?

**Comment by jianhandev** [comment]
Hi @Jellybeano make sure you have followed the setup on [this page](https://github.com/TEAMMATES/teammates/blob/master/docs/setting-up.md) before running `gradlew.bat serverRun`. 

The full-text search service is optional unless you are working on search-related features. If `runDatastoreEmulator` does not work for you, you can try the other method [using Cloud SDK](https://github.com/TEAMMATES/teammates/blob/master/docs/development.md#using-cloud-sdk).

**Reply by Jellybeano** [reply]
Hi thanks for the help!

All setup steps have already been completed, including installing JDK 1.8, Python 3, Node.js, Angular CLI, the gradle.properties file as well as the node_modules folder are both present. (unfortunately)

Managed to get the Google Cloud datastore to work:
![image](https://user-images.githubusercontent.com/65604398/122970205-2ef63100-d3c0-11eb-87d8-7d53b0224b23.png)

However the same error message still persists when running gradlew.bat serverRun with HTTP ERROR 404 Not Found at http://localhost:8080/

Any thoughts on why this problem occurs?

**Reply by daongochieu2810** [reply]
@Jellybeano  can you provide the content of `gradle.properties`?

**Reply by Jellybeano** [reply]
yup sure, the contents are pasted below! (#'s removed for formatting purposes)

Use this property to enable/disable Gradle Daemon (long-lived background process)
Reference: https://docs.gradle.org/current/userguide/gradle_daemon.html
org.gradle.daemon=false

Use this property if you want to use a JDK other than the one specified in your PATH variable.
Windows users should use a forward slash (/) instead of the Windows default backward slash (\) while specifying the path.
e.g org.gradle.java.home=/Library/Java/JavaVirtualMachines/jdk1.8.0_202.jdk/Contents/Home
e.g org.gradle.java.home=C:/Program Files/Java/jdk1.8.0_202
org.gradle.java.home=C:/Users/Lan Yu Xuan/.jdks/corretto-1.8.0_292

Use this property if you want to use a specific Cloud SDK installation.
If this property is not set, the latest Cloud SDK will always be used.
cloud.sdk.home=/Users/visenze/Documents/google-cloud-sdk

**Reply by damithc** [reply]
> yup sure, the contents are pasted below! (#'s removed for formatting purposes)

@Jellybeano You can also use fenced code blocks to prevent formatting being applied

e.g.,
```
#Use this property to enable/disable Gradle Daemon (long-lived background process)
#Reference: https://docs.gradle.org/current/userguide/gradle_daemon.html
org.gradle.daemon=false

#Use this property if you want to use a JDK other than the one specified in your PATH variable.
#Windows users should use a forward slash (/) instead of the Windows default backward slash () while specifying the path.
#e.g org.gradle.java.home=/Library/Java/JavaVirtualMachines/jdk1.8.0_202.jdk/Contents/Home
#e.g org.gradle.java.home=C:/Program Files/Java/jdk1.8.0_202
org.gradle.java.home=C:/Users/Lan Yu Xuan/.jdks/corretto-1.8.0_292

#Use this property if you want to use a specific Cloud SDK installation.
#If this property is not set, the latest Cloud SDK will always be used.
cloud.sdk.home=/Users/visenze/Documents/google-cloud-sdk
```

**Comment by daongochieu2810** [comment]
@Jellybeano I feel like you got the wrong jdk version, can you try downloading from this https://www.oracle.com/sg/java/technologies/javase/javase-jdk8-downloads.html

**Reply by Jellybeano** [reply]
oh alright thanks!

**Comment by yixingqian** [comment, marked as answer]
@Jellybeano Hey
Can you try to remove any space in your project path and see if it works.

**Reply by Jellybeano** [reply]
i cannot believe this was why the program was breaking, thanks so much!! 🙏
